### PR TITLE
Add a new test to verify python311_virtualenv

### PR DESCRIPTION
**Pull request by DeepthiYV**
- Related ticket:  https://progress.opensuse.org/issues/156472
- Needles: NO
- Verification run: 
 SLE15-SP6 : [aarch64](https://openqa.suse.de/tests/14862059) | [s390x](https://openqa.suse.de/tests/14862061) | [x86_64](https://openqa.suse.de/tests/14862068)
SLE15-SP4:  [aaarch64](https://openqa.suse.de/tests/14862014) | [s390x](https://openqa.suse.de/tests/14862017) | [x86_64](https://openqa.suse.de/tests/14862025)
SLE15-SP5:  [aaarch64](https://openqa.suse.de/tests/14862001) | [s390x](https://openqa.suse.de/tests/14861999) | [x86_64](https://openqa.suse.de/tests/14862000)
Tumbleweed:  https://openqa.opensuse.org/tests/4343074
LEAP 15.6 : [x86_64](https://openqa.opensuse.org/tests/4343078) | [ppc64le](https://openqa.opensuse.org/tests/4343077) | [aarch64](https://openqa.opensuse.org/tests/4343076)

### CODE REVIEW

File: tests/console/python_virtualenv.pm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+no warnings qw(experimental::signatures);
+
+sub run {
+    select_serial_terminal;
+    # Import the project directory for creating a source distribution package.
+    # The directory should contain setup.py and a Python module named test_module.py
+    assert_script_run('curl -L -s ' . data_url('python/python3-setuptools') . ' | cpio --make-directories --extract && cd data');
+    # Install python311-virtualenv
+    if (zypper_call("se -x python311-pipx", exitcode => [0, 104]) == 104) {
+        die("python311-pipx doesnt exist");
```
Comment (rfan1) [inline tests/console/python_virtualenv.pm:37]: s/doesnt/doesn't/

File: tests/console/python_virtualenv.pm [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+sub post_run_hook {
+    assert_script_run("deactivate");
+    zypper_call('rm python311-pipx', exitcode => [0, 104]);
+    zypper_call('rm python311-base', exitcode => [0, 104]);
+}
+
+sub post_fail_hook {
+    assert_script_run("deactivate");
```
Comment (rfan1) [inline tests/console/python_virtualenv.pm:59]: I would suggest to  use `script_run` :)

File: tests/console/python_virtualenv.pm [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    # create a virtual environment named myenv using virtualenv with Python 3.11 and activate it
+    assert_script_run("pipx run --python=python3.11 virtualenv myenv");
+    assert_script_run("source myenv/bin/activate");
+    # Install build tools and build, install the package locally
+    assert_script_run("pip install setuptools wheel");
+    assert_script_run("python setup.py sdist bdist_wheel");
+    validate_script_output("pip install dist/user_package_setuptools-1.0-py3-none-any.whl", sub { m/Successfully installed/ });
+    # Verify the installed package
+    validate_script_output('pip list', sub { m/user_package_setuptools/ });
+    assert_script_run("python3.11 sample_test_module/test_module.py");
```
Comment (foursixnine) [inline tests/console/python_virtualenv.pm:49]: Are we sure that in this case, is pipx's pip and python?

Comment (DeepthiYV) [reply tests/console/python_virtualenv.pm:49]: I will automate another module to focus on testing pipx. Pipx manages Python applications and tools in isolated environments.

**Comment by foursixnine** [comment]
@DeepthiYV What is missing here?

**Review by foursixnine — APPROVED** [review]

**Review by rfan1 — APPROVED** [review]

**Pull request merged**